google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/AppEngineFacetConfiguration.java

```diff
@@ -69,19 +71,11 @@ public void writeExternal(Element element) throws WriteExternalException {
    * returned and therefore not considered to be "managed".
    */
   public Set<AppEngineStandardMavenLibrary> getLibraries(@NotNull Project project) {
-    LibraryTable libraryTable = LibraryTablesRegistrar.getInstance().getLibraryTable(project);
-    Library[] libraries =  libraryTable.getLibraries();
-
-    Set<AppEngineStandardMavenLibrary> configuredLibraries = Sets.newHashSet();
-    for (Library configuredLibrary : libraries) {
-      AppEngineStandardMavenLibrary mavenLibrary
-          = AppEngineStandardMavenLibrary.getLibraryByMavenDisplayName(configuredLibrary.getName());
-      if (mavenLibrary != null) {
-        configuredLibraries.add(mavenLibrary);
-      }
-    }
-
-    return configuredLibraries;
+    return Arrays.stream(
+        LibraryTablesRegistrar.getInstance().getLibraryTable(project).getLibraries())
+        .map(library ->
+            AppEngineStandardMavenLibrary.getLibraryByMavenDisplayName(library.getName()).get())
```

Comment:
What I think happens is map maps to Optional.empty(), in case of empty optionals returned by the mapping method (i.e., no nulls involved), and then later, the collector skips empty optionals.

The example in http://www.oracle.com/technetwork/articles/java/java8-optional-2175753.html seems to indicate that, and I think the Java 8 in Action book covers that, too.